src/uniswap-summaries.md

```diff
@@ -2890,6 +2890,30 @@ module SOLIDITY-UNISWAP-FIDUPDATE-3-SUMMARY
 endmodule
 ```
 
+```k
+module SOLIDITY-UNISWAP-GETRESERVES-SUMMARY 
+  imports SOLIDITY-CONFIGURATION
+  imports SOLIDITY-EXPRESSION
+  imports SOLIDITY-UNISWAP-TOKENS
+
+  rule <k> bind ( _STORE , .List , .List , .TypedVals , ListItem ( uint112 []:TypeName ) , ListItem ( reserves ) ) ~> reserves = new uint112 [ ] ( 3 , .TypedVals ) ;  reserves [ 0 ] = reserve0 ;  reserves [ 1 ] = reserve1 ;  reserves [ 2 ] = blockTimestampLast ;  .Statements ~> return reserves ; ~> .K => return v ( ListItem({Storage[reserve0] orDefault 0p112}:>MInt{112}) ListItem({Storage[reserve1] orDefault 0p112}:>MInt{112}) ListItem(roundMInt({Storage[blockTimestampLast] orDefault 0p32}:>MInt{32}):MInt{112}) , uint112 [ ] ) ; ~> .K</k> 
+        <summarize> true </summarize>
+        <this> THIS </this>
+        <contract-address> THIS </contract-address>
+        <this-type> TYPE </this-type>
+        <contract-id> TYPE </contract-id>
+        <current-function> getReserves </current-function>
+        <env> ENV=> ENV [ reserves <- var(1, uint112 []) ] </env>
```

Comment:
It works with both `0` or `1` (dind't tried other numbers). That's my concern...
In a execution that I was tracking we had:
- Before 1st bind: `env: .Map`
- After 1st bind: `env: reserves |-> (0, uint112 [])`
- After 2nd bind:  `env: reserves |-> (0, uint112 [])`
- Body execution...

Using `size(S)` it returns 0, as before the `bind` we had a empty map. However, the function behavior is to have `1` there, as we have 1 element just added. So, I would suggest `size(S) + 1`. What do you think?

Reply:
The first bind is for the arguments, and there are none. The second is for the return.
Did you run the function without this, starting from right after the call? That will show you what the store size is. Then, you just pick the next number, which is size(S).

The explanation I would give to this is that this is from an external call, which rewrites the Store to .List. Therefore, its current size is 0, and this is the number that is selected.
There is a coincidence that makes it work correctly: the next instruction is a new (creates an item in store, increasing its size by one) that is then assigned to reserves, so making it go from 0 to 1. I think it works with size(S) because it is the correct thing to do, and it worked with 1 because of this coincidence.

Reply:
Also, it is my understanding that `size(S-1)` is the last element. `size(S)` would be a new element just added, as part of the rewrite. I say this because list indices start from 0.

Reply:
Alright!